### PR TITLE
feat: State constructor with schema

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
When an nns-dapp canister is deployed to an empty canister or redeployed, we would like to be able to use any specified schema.

# Changes
- Add a  constructor for state that takes memory and schema as arguments.

# Tests
- A test is provided that verifies that the state is indeed created with the requested schema.

# Todos

- [ ] Add entry to changelog (if necessary).

### CODE REVIEW

File: rs/backend/src/state.rs [hunk 1]
```diff
@@ -96,6 +103,41 @@ thread_local! {
...
 
+impl State {
+    /// Creates new state with the specified schema.
+    #[cfg(test)]
+    pub fn new(schema: SchemaLabel, memory: DefaultMemoryImpl) -> Self {
+        use crate::{accounts_store::schema::AccountsDbTrait, state::partitions::PartitionType};
+
+        match schema {
+            SchemaLabel::Map => {
+                println!("New State: Map");
```
Comment (przydatek) [inline rs/backend/src/state.rs:114]: nit: are these log messages (here and in line 124) intentional?

File: rs/backend/src/state/tests.rs [hunk 2]
```diff
@@ -45,3 +45,41 @@ fn schema_can_be_read_from_memory() {
...
+        state
+            .accounts_store
+            .borrow_mut()
+            .db_insert_account(&[0u8; 32], crate::accounts_store::schema::tests::toy_account(1, 2));
+
+        // Do the partitions look as expected?
+        let partitions_maybe_label = format!("{}", state.partitions_maybe.borrow());
+        match (schema, partitions_maybe_label.as_str()) {
+            (SchemaLabel::Map, "None") | (SchemaLabel::AccountsInStableMemory, "Partitions") => {
+                println!("Partitioning is as expected for {schema:?}!");
```
Comment (przydatek) [inline rs/backend/src/state/tests.rs:71]: nit: also here, should the `println` remain?

**Review by przydatek — APPROVED** [review]
LGTM, thanks!

File: rs/backend/src/state/tests.rs (hunk 2 above)
Comment (bitdivine) [reply rs/backend/src/state/tests.rs:71]: This one is also intentional for the same reason. :-)

File: rs/backend/src/state.rs (hunk 1 above)
Comment (bitdivine) [reply rs/backend/src/state.rs:114]: Resolving the conversation so that I can merge but happy to discuss further or make changes later if wanted.

**Pull request merged**